crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -290,6 +294,7 @@ impl Error {
             }
             Error::GasPrice(_) => false,
             Error::AccessList(_) => true,
+            Error::UnsupportedChain => false,
```

Comment:
A new error variant gets added which can never appear here and yet we have to handle this case. This is a code smell and indicates that these functions should return different errors.
Something for a future refactor, though.

Reply:
I don't have a strong opinion here. If you look at how the same constructor is used in `autopilot`, you can see that `.expect()` is used when the `Chain::try_from` is called in `run.rs` (so when we are aware of our surrounding and we explicitly want to exit), and error type is used inside `Rpc::new` because we don't know when and how this function will be called.

But I also agree that this error is rarely hit. If we agree to panic everywhere, then we can remove `ChainIdNotSupported` error from `chain` crate altogether. This is dangerous since we might want to try to build the `Chain` object from unreliable source that could give garbage, but it's not the case now.

Reply:
So another option would be to introduce new error type dedicated for `Rpc` object, for instance `RpcError`, which will contain only one variant `ChainIdNotSupported`, and caller of the constructor will decide if want to panic or map error to other type.
Alternatively constructor can return boolean type error as there is only one function which can fail with one error code.

Reply:
> So another option would be to introduce new error type dedicated for Rpc object, for instance RpcError, which will contain only one variant ChainIdNotSupported, and caller of the constructor will decide if want to panic or map error to other type.

That seems reasonable. @sunce86 do you mind implementing it?

Reply:
Sure, I created a new error type